diff --git a/ios/Core/SharedObjects/SharedObjectRegistry.swift b/ios/Core/SharedObjects/SharedObjectRegistry.swift
--- a/ios/Core/SharedObjects/SharedObjectRegistry.swift
+++ b/ios/Core/SharedObjects/SharedObjectRegistry.swift
@@ -34,7 +34,7 @@ public final class SharedObjectRegistry: Sendable {
   /**
    Weak reference to the app context for the registry.
    */
-  private weak let appContext: AppContext?
+  nonisolated(unsafe) private weak var appContext: AppContext?
 
   internal struct State: Sendable {
     /**
