diff --git a/src/components/bottomSheet/BottomSheet.tsx b/src/components/bottomSheet/BottomSheet.tsx
index 6748c93cac1a29c745480fe2282b8d8f6b36226f..b10b4d461eef4f25b29546eaf9af6b5a872e6768 100644
--- a/src/components/bottomSheet/BottomSheet.tsx
+++ b/src/components/bottomSheet/BottomSheet.tsx
@@ -1709,13 +1709,14 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
          }
 
          /**
-         * if new keyboard state is hidden and blur behavior is none, then exit the method
+         * if new keyboard state is hidden and blur behavior is none,
+         * clear temporary position so evaluatePosition can snap to detent.
           */
          if (
            status === KEYBOARD_STATUS.HIDDEN &&
            keyboardBlurBehavior === KEYBOARD_BLUR_BEHAVIOR.none
          ) {
-          return;
+          isInTemporaryPosition.value = false;
          }
 
          const animationConfigs = getKeyboardAnimationConfigs(easing, duration);
diff --git a/src/components/bottomSheetTextInput/BottomSheetTextInput.tsx b/src/components/bottomSheetTextInput/BottomSheetTextInput.tsx
index 861d276ec59022d8c9de64cc0920972f68d4a6a4..2dd7b7349b19f3180cd0a407071029907643e06e 100644
--- a/src/components/bottomSheetTextInput/BottomSheetTextInput.tsx
+++ b/src/components/bottomSheetTextInput/BottomSheetTextInput.tsx
@@ -63,6 +63,7 @@ const BottomSheetTextInputComponent = forwardRef<
          animatedKeyboardState.set(state => ({
            ...state,
            target: undefined,
+          height: 0,
          }));
        }
 
@@ -100,6 +101,7 @@ const BottomSheetTextInputComponent = forwardRef<
          animatedKeyboardState.set(state => ({
            ...state,
            target: undefined,
+          height: 0,
          }));
        }
 
